torch/utils/data/sampler.py

```diff
@@ -346,3 +347,57 @@ def __len__(self) -> int:
             return len(self.sampler) // self.batch_size  # type: ignore[arg-type]
         else:
             return (len(self.sampler) + self.batch_size - 1) // self.batch_size  # type: ignore[arg-type]
+
+class RandomBatchSampler(Sampler):
+    def __init__(
+        self,
+        data_source: Sized,
+        batch_size: int,
+        drop_last: bool,
+        replacement: bool = False,
+        generator=None,
+    ) -> None:
+        # Validate parameters using init of RandomSampler and BatchSampler
+        RandomSampler(data_source, replacement, None, generator)
+        BatchSampler([], batch_size, drop_last)
+
+        self.data_source = data_source
+        self.batch_size = batch_size
+        self.drop_last = drop_last
+        self.replacement = replacement
+        self.generator = generator
+
+        self.n_batches = len(self.data_source) // self.batch_size
+        last_size = len(self.data_source) % self.batch_size
+        if not self.drop_last and last_size > 0:
+            self.n_batches += 1
+
+
+    def sample_indices(self) -> torch.Tensor:
+        n = len(self.data_source)
+        generator = self.generator if self.generator is not None else init_generator()
+
+        if self.replacement:
+            indices = torch.randint(high=n, size=(n,), dtype=torch.int64, generator=generator)
+        else:
+            indices = torch.randperm(n, generator=generator)
+
+        return indices
+
+    def __iter__(self) -> Iterator:
+        indices = self.sample_indices()
+
+        # Slicing is faster on list when batch size is small
+        if self.batch_size < 16:
+            indices = indices.tolist()  # type: ignore
```

Comment:
In case `batch_size < 16` the yielded type is `list[int]`, otherwise it is `Tensor`.
Is this an issue?